model-optimizer/extensions/ops/select.py

```diff
@@ -39,28 +38,48 @@ def infer(node: Node):
 
         a_shape = node.in_port(1).data.get_shape()
         b_shape = node.in_port(2).data.get_shape()
-        output_shape = bi_directional_shape_broadcasting(a_shape, b_shape)
+        broadcast_rule = node.soft_get('auto_broadcast', 'numpy')
+        if broadcast_rule == 'numpy':
+            output_shape = bi_directional_shape_broadcasting(a_shape, b_shape)
+        elif broadcast_rule == 'pdpd':
+            # todo: add pdpd broadcasting rule
+            # note that additionally to output_shape resulting_tensors must be broadcasted as well
+            raise Error("PDPD broadcasting rule is not implemented yet")
+        else:  # broadcasting is not allowed
+            assert np.array_equal(a_shape, b_shape) and condition_value is not None and np.array_equal(condition_value.shape, a_shape), \
+                'In node \'{}\' for Select operation when broadcasting is off all inputs must be of the same shape. ' \
+                'But instead got: cond_shape={}, then_shape={}, else_shape={}'.format(
+                    node_name, condition_value.shape, a_shape, b_shape)
+            output_shape = a_shape
```

Comment:
In case some shape has dynamic dimension and the other one has fully defined then you need to use the fully defined dimension(, assuming that dynamic dimension must be equal).
Please test cases for it